scripts/modules/apm_agents.py

```diff
@@ -89,9 +87,7 @@ def __init__(self, **options):
         self.agent_version = options.get("go_agent_version", self.DEFAULT_AGENT_VERSION)
         self.agent_repo = options.get("go_agent_repo", self.DEFAULT_AGENT_REPO)
         if options.get("enable_apm_server", True):
-            self.depends_on = {
```

Comment:
removing this will not wait for the APM Server healthcheck, this means that agents can start to send data to APM Server when it is not running. Is there a reason?

Reply:
I think I had experienced a problem while migrating to the v3 version of Docker Compose, but I just tested it and it seems to work fine with the long syntax so I'm going to revert this set of changes. Thanks for pointing this out!

Reply:
@cachedout , we cannot use docker-compose v3 since from >2.4, if I recall correctly, there is no support for `service_healthy`.

Reply:
@v1v I [read the spec ](https://github.com/compose-spec/compose-spec/blob/master/spec.md#long-syntax-1)this morning and it appears to be there. I've been testing with it so far today and it has been working fine as I 've been going through and doing the various reverts. I'm not completely sure why this wasn't working before, to be honest.

Reply:
Reverted this change and restored the `depends_on`. Resolving thread.